.github/workflows/build-wheel-linux-arm64.yaml

```
@@ -37,291 +34,415 @@ jobs:
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/mlir/Enzyme
+          key: enzyme-${{ needs.constants.outputs.enzyme_version }}-container-source
+          enableCrossOsArchive: True
+
+      - name: Clone LLVM Submodule
+        if: steps.cache-llvm-source.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: llvm/llvm-project
+          ref: ${{ needs.constants.outputs.llvm_version }}
+          path: ${{ github.workspace }}/mlir/llvm-project
+
+      - name: Clone MHLO Submodule
+        if: steps.cache-mhlo-source.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: tensorflow/mlir-hlo
+          ref: ${{ needs.constants.outputs.mhlo_version }}
+          path: ${{ github.workspace }}/mlir/mlir-hlo
+          
+      - name: Clone Enzyme Submodule
+        if: steps.cache-enzyme-source.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: EnzymeAD/Enzyme
+          ref: ${{ needs.constants.outputs.enzyme_version }}
+          path: ${{ github.workspace }}/mlir/Enzyme                  
+
+
+      # Cache external project builds
+      - name: Restore LLVM Build
+        id: cache-llvm-build
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/llvm-build
+          key: ${{ matrix.container_img }}-llvm-${{ needs.constants.outputs.llvm_version }}-${{matrix.python_version}}-wheel-build
+
+      - name: Restore MHLO Build
+        id: cache-mhlo-build
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/mhlo-build
+          key: ${{ matrix.container_img }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-wheel-build
+          lookup-only: True
+
+      - name: Restore Enzyme Build
+        id: cache-enzyme-build
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/enzyme-build
+          key: ${{ matrix.container_img }}-enzyme-${{ needs.constants.outputs.llvm_version }}-${{ needs.constants.outputs.enzyme_version }}-wheel-build
+          lookup-only: True
+
+      - name: Install dependencies
+        if: |
+          steps.cache-llvm-build.outputs.cache-hit != 'true' ||
+          steps.cache-mhlo-build.outputs.cache-hit != 'true' ||
+          steps.cache-enzyme-build.outputs.cache-hit != 'true'
+        run: |
+          # echo /Library/Frameworks/Python.framework/Versions/${{ matrix.python_version }}/bin >> $GITHUB_PATH
+          cat /etc/dnf.conf | sed "s/\[main\]/\[main\]\ntimeout=5/g" > /etc/dnf.conf
+          dnf update -y && dnf install -y libzstd-devel gcc-toolset-12 
+          # Update env vars valid for all tasks of this job
+          echo '/opt/rh/gcc-toolset-12/root/usr/bin' >> $GITHUB_PATH          
+
+      - name: Set up Python ${{ matrix.python_version }}
+        run: |
+          dnf install -y python3
```

Is this the way to setup python on the Linux ARM runners? ... as in, we are not able to use actions/setup-python?

```suggestion
      - name: Set up Python ${{ matrix.python_version }}
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python_version }}
```

---

I'm pretty sure Python will be included on the manylinux image, is this entire step not obsolete?

---

i'll remove it. it should still run, the workflow had trouble finding python earlier

---

nvm [keeping](https://github.com/PennyLaneAI/catalyst/actions/runs/14524233140/job/40752637248#step:5:5) it in